src/Security/MemberAuthenticator/ChangePasswordHandler.php

```diff
@@ -348,4 +387,48 @@ protected function checkPassword($member, $password)
         }
         return true;
     }
+
+    /**
+     * Generate a temporary auto login hash for the member
+     */
+    private function createAutoLoginTempHash(): string
+    {
+        if (is_callable($this->autoLoginTempHashGenerator)) {
+            return call_user_func($this->autoLoginTempHashGenerator);
+        }
+        return $this->createDefaultAutoLoginTempHash();
+    }
+
+    /**
+     * Generate a temporary auto login hash for the member
+     */
+    private function createDefaultAutoLoginTempHash(): string
+    {
+        $member = null;
+        $autoLoginTempHash = '';
+        do {
+            $autoLoginTempHash = hash('sha256', random_int(0, PHP_INT_MAX));
+            $member = Member::get()->find('AutoLoginTempHash', $autoLoginTempHash);
+        } while ($member !== null);
+        return $autoLoginTempHash;
+    }
```

Comment:
Just looked at how the `` is created - looks like we already have an API for making these hashes which we should probably just reuse here.

```php
$generator = RandomGenerator::create();
$token = $generator->randomToken();
```

`RandomGenerator` is even injectable which resolves the other discussion about having a property and setter vs mocking etc - just set a mock generator in the injector.

Reply:
Have updated to use RandomGenerator, though it's not injectable